repl/src/main/scala/ammonite/repl/Main.scala

```diff
@@ -70,21 +80,104 @@ case class Main(predef: String = "",
     res
   }
 
+
+  def runScriptLevelCache(path: Path, args: Seq[String] = Vector.empty,
+                          kwargs: Map[String, String] = Map.empty,
+                          storage: Storage) = {
+    Timer.startTime = System.nanoTime()
+    val code: String = read(path)
+    val (pkg, wrapper) = Util.pathToPackageWrapper(path, wd)
+    val cacheTag = "cache" + Util.md5Hash(Iterator(code.getBytes)).map("%02x".format(_)).mkString
+    // check if script takes args or loads code as they will need compiler
+    storage match {
+      case s: Storage.InMemory => runScript(path, args, kwargs, storage, cacheTag, false)
+      case _ =>
+        if (args == Seq() || !code.contains("load(")) {
+          storage.asInstanceOf[Storage.Folder].classFilesListLoad(pkg, wrapper, cacheTag) match {
+            case Seq() =>
+              runScript(path, args, kwargs, storage, cacheTag)
+            case cachedData =>
+              val outBuffer = mutable.Buffer.empty[String]
+              val warningBuffer = mutable.Buffer.empty[String]
+              val errorBuffer = mutable.Buffer.empty[String]
+              val infoBuffer = mutable.Buffer.empty[String]
```

Comment:
Not sure why we're building up these buffers and throwing them away, but it doesn't matter anyway if we're getting rid of `runScriptLevelCache` as described in my high-level feedback